src/file_lines.rs

```diff
@@ -0,0 +1,221 @@
+//! This module contains types and functions to support formatting specific line ranges.
+use std::{cmp, iter, str};
+
+use itertools::Itertools;
+use multimap::MultiMap;
+use rustc_serialize::{self, json};
+
+use codemap::LineRange;
+
+/// A range that is inclusive of both ends.
```

Comment:
I added a note in the README about them being 1-based.

For closed vs half-open ranges, I can try to do a survey of text editors and how they represent the line range of a selection for sending to an external command. Are you ok with merging this as-is, and revisiting after the survey?